src/main/scala/scalaz/stream/nio/file.scala

```diff
@@ -157,6 +157,39 @@ object file {
       }
   }
 
+  private def withIsDirectory(path:Path): Task[(Path, Boolean)] = 
+    Task.delay(path -> Files.isDirectory(path))
+
+  private val isDirectoryW = Process.constant(withIsDirectory _).toSource
```

Comment:
this can be expressed as `sink.lift { Task.delay(path -> Files.isDirectory(path))   }`, but really I think just doing the test in eval() and flatMap has better performance here.